src/ccs/multifolding.rs

```diff
@@ -56,29 +58,51 @@ impl<G: Group> NIMFS<G> {
       ccs_mle,
       ck,
       lcccs,
+      transcript: TranscriptEngineTrait::new(b"NIMFS"),
     }
   }
 
   /// Initializes a NIMFS instance given the CCS of it and a first witness vector that satifies it.
   // XXX: This should probably return an error as we should check whether is satisfied or not.
-  pub fn init<R: RngCore>(mut rng: &mut R, ccs: CCS<G>, z: Vec<G::Scalar>) -> Self {
+  pub fn init(ccs: CCS<G>, z: Vec<G::Scalar>, seed: &'static [u8]) -> Self {
```

Comment:
`seed` makes it sound like it expects an rng output? should this be named `label`?